File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    wall=[False]*n
    for i in section: wall[i-1]=True
    for i,paint in enumerate(wall):
        if paint==True:
            for j in range(m): 
                if i+j< n: 
                    wall[i+j]=False
            answer+=1
        
    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
import java.util.*;
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        StringBuilder sb=new StringBuilder("o".repeat(n));
        for(int w:section) sb.setCharAt(w-1,'x');
        String wall=sb.toString();
        
        while(wall.indexOf('x')!=-1){
             int index=wall.indexOf('x');
             if(index>=n-m){
                 answer++;
                 break;
             }
             wall=wall.substring(0,index)+"o".repeat(m)+wall.substring(index+m);
             answer++;
         }
        
        // for(int w:section){
        //     if(wall.charAt(w-1)=='x'){
        //         if(w<n-m+1) wall.replace(w-1,w+m-1,"o".repeat(m));
        //         answer++;
        //     }
        // }
        
        return answer;
    }
}